src/components/EnrolmentVillagesPicker.js

```diff
@@ -66,20 +69,77 @@ const EnrolmentVillagesPicker = (props) => {
   const onInsertRow = () => {
     setItems([...items, {}]);
   };
+
   const onSelectParent = (item, parent) => {
-    // eslint-disable-next-line no-param-reassign
-    item.parent = parent;
+    const rowItem = item;
+    rowItem.parent = parent;
     setItems([...items]);
   };
+
   const onVillagesChange = (item, entities) => {
-    // eslint-disable-next-line no-param-reassign
-    item.entities = entities;
+    const rowItem = item;
+    rowItem.entities = entities;
     handleChange(items);
   };
 
-  // Only display options for municipalities not already present.
+  const createRow = (parent) => {
+    const {
+      children: { edges },
+    } = parent;
+    const entities = edges?.map((edge) => edge.node) ?? [];
+    const createdRow = {};
+
+    createdRow.parent = parent;
+    createdRow.entities = entities;
+    return createdRow;
+  };
+
+  const clearItems = () => {
+    setItems([]);
+    handleChange([]);
+  };
+
+  const executeNewRows = (newRows, uniqueRows) => {
+    if (!items.length) {
+      setItems([...items, ...newRows]);
+      handleChange([...items, ...newRows]);
+    } else {
+      setItems([...items, ...uniqueRows]);
+      handleChange([...items, ...uniqueRows]);
+    }
+  };
+
   const filterParents = (options) => options.filter((p) => !items.some((i) => i.parent?.id === p.id));
 
+  const filterVillages = (options) =>
+    options.filter((option) => !villages?.some((village) => village?.id === option.id));
+
+  useEffect(() => {
+    if (districts?.length) {
+      clearItems();
```

Comment:
seems that 119-120, 138-139 and 123-124 lines are repeatable. Can we create separate function for that?

Reply:
Instead of doing seperated function, I used `clearItems` function and inside it I'm calling the `dispatch(clearDistrictData())`.